app.py

```diff
@@ -344,3 +361,44 @@ def cancel_carpool(carpool_id):
             return redirect(url_for('carpool_details', carpool_id=carpool_id))
 
     return render_template('cancel_carpool.html', form=cancel_form)
+
+
+def _email_carpool_cancelled(carpool_id, reason, send=False):
+    carpool = Carpool.query.get(int(carpool_id))
+    driver = Person.query.get(int(carpool.driver_id))
+    riders = carpool.riders
+    if len(riders) == 0:
+        return
+
+    if not reason:
+        reason = 'Reason not given!'
+
+    subject = 'Carpool session on {} cancelled'.format(carpool.leave_time)
+
+    body = '''
```

Comment:
We might not want to do it in this pull request, but these emails should probably come from templates.